services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -174,6 +177,97 @@ class FlowHelper {
         } || existingFlows*.id.contains(newFlow.id)
     }
 
+    /**
+     * Check that all needed rules are created for a flow with protected path
+     *
+     * @param flowId
+     */
+    void verifyRulesOnProtectedFlow(String flowId) {
+        // TODO (andriidovhan) add possibility to use this method when we have more than 1 flow on a switch
+        // then update addFlow/deleteFlow/updateFlow
+        def mainFlowPath = northbound.getFlowPath(flowId).forwardPath
+        def srcMainNode = mainFlowPath[0]
+        def dstMainNode = mainFlowPath[-1]
+        def mainFlowTransitNodes = mainFlowPath[1..-2]
+        def protectedFlowPath = northbound.getFlowPath(flowId).protectedPath.forwardPath
+        def srcProtectedNode = protectedFlowPath[0]
+        def dstProtectedNode = protectedFlowPath[-1]
+        def protectedFlowTransitNodes = protectedFlowPath[1..-2]
+
+        def commonNodes = mainFlowPath*.switchId.intersect(protectedFlowPath*.switchId)
+        def commonTransitSwitches = mainFlowTransitNodes*.switchId.intersect(protectedFlowTransitNodes*.switchId)
+
+        def rulesOnSrcSwitch = northbound.getSwitchRules(srcMainNode.switchId).flowEntries.findAll {
+            !Cookie.isDefaultRule(it.cookie)
+        }
+        assert rulesOnSrcSwitch.size() == 3
+        //rules for main path
+        assert findInputRules(rulesOnSrcSwitch, srcMainNode).size() == 1
+        assert findOutputRules(rulesOnSrcSwitch, srcMainNode).size() == 1
+        //rule for protected path
+        assert findInputRules(rulesOnSrcSwitch, srcProtectedNode).size() == 0
+        assert findOutputRules(rulesOnSrcSwitch, srcProtectedNode).size() == 1
+
+        def rulesOnDstSwitch = northbound.getSwitchRules(dstMainNode.switchId).flowEntries.findAll {
+            !Cookie.isDefaultRule(it.cookie)
+        }
+        assert rulesOnDstSwitch.size() == 3
+        //rules for main path
+        assert findInputRules(rulesOnDstSwitch, dstMainNode).size() == 1
+        assert findOutputRules(rulesOnDstSwitch, dstMainNode).size() == 1
+        //rule for protected path
+        //dst switch contain egress rule for protected, but it looks like ingress
+        assert findInputRules(rulesOnDstSwitch, dstProtectedNode).size() == 1
+        assert findOutputRules(rulesOnDstSwitch, dstProtectedNode).size() == 0
+
+        if (commonTransitSwitches) {
+            commonTransitSwitches.each { sw ->
```

Comment:
Add descriptive comment for this block

Reply:
done
also description of method was updated